File: api/src/endpoint/oauth/create_device_code.rs
use api_derive::ApiRequest;
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::oauth::DeviceCodeResource};

pub struct CreateDeviceCode;

impl Endpoint for CreateDeviceCode {
    const PATH: &'static str = "/oauth/device";
    const METHOD: http::Method = http::Method::POST;

    type Request = CreateDeviceCodeRequest;
    type Response = CreateDeviceCodeResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct CreateDeviceCodeRequest {
    pub client_id: String,
}

pub type CreateDeviceCodeResponse = DeviceCodeResource;

File: gitdot-api-ts/src/endpoint/oauth/create-device-code.ts
import { z } from "zod";
import { DeviceCodeResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const CreateDeviceCodeRequest = z.object({
  client_id: z.string(),
});
export type CreateDeviceCodeRequest = z.infer<typeof CreateDeviceCodeRequest>;

export const CreateDeviceCodeResponse = DeviceCodeResource;
export type CreateDeviceCodeResponse = z.infer<typeof CreateDeviceCodeResponse>;

export const CreateDeviceCode = {
  path: "/oauth/device",
  method: "POST",
  request: CreateDeviceCodeRequest,
  response: CreateDeviceCodeResponse,
} as const satisfies Endpoint;

export type CreateDeviceCode = typeof CreateDeviceCode;
